src/Tinker.jl

```diff
@@ -1,4 +1,82 @@
 module Tinker
 
 # package code goes here
+
+using Gtk.ShortNames, GtkReactive, TestImages, Graphics, Colors, Images
+
+function setupfile(file)
+    # file info
+    image = load(file)
+
+    # set up window
+    win = Window(file, size(image,2), size(image,1)); #name of image
+    c = canvas(UserUnit);
+    push!(win, c);
+
+    # set up a zoom region
+    zr = Signal(ZoomRegion(image))
+
+    # create view
+    imagesig = map(zr) do r
+        cv = r.currentview
+        view(image, UnitRange{Int}(cv.y), UnitRange{Int}(cv.x))
+    end;
+    
+    # create a view diagram
+    viewdim = map(zr) do r
+        fvx = r.fullview.x # x range of image
+        fvy = r.fullview.y # y range if image
+        cvx = r.currentview.x # the x range of currentview
+        cvy = r.currentview.y # the y range of currentview
+        xsc = (cvx.right-cvx.left)/(10*(fvx.right-fvx.left)) # x scale
```

Comment:
Just to clarify: the field you're worried about in this case is .right and .left - .currentview & .x/.y aren't going to change and I shouldn't worry about getting them through some other means?